environments/dev-env/install_req.sh

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env bash
+# Copyright 2023 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+get_os_type() {
+  case "$OSTYPE" in
+      darwin*)  echo "darwin" ;;
+      linux*)   echo "linux" ;;
+      *)        echo "unknown: $OSTYPE" && exit 1 ;;
+  esac
+}
+
+os_type=$(get_os_type)
+
+# Install golang for x86_64
+if [ $os_type == "linux" ]; then
+    if !(command -v go) then
+        sudo apt update
+        sudo apt upgrade -y
+        curl -O https://dl.google.com/go/go1.19.2.linux-amd64.tar.gz
+        tar -xvf go1.19.2.linux-amd64.tar.gz
+        sudo mv go  /usr/bin/
+        sudo echo 'export GOROOT=/usr/bin/go' >> ~/.bashrc 
+        sudo echo 'export PATH=$PATH:$GOROOT/bin' >> ~/.bashrc 
+        sleep 5
+        exec bash
+    else
+        echo "Go is already installed"
+        echo "***********************"
+    fi
+elif [ $os_type == "darwin" ]; then
+    if !(command -v go) then
+        brew update
+        curl -O https://dl.google.com/go/go1.19.2.darwin-amd64.pkg
+        tar -xvf go1.19.2.darwin-amd64.pkg
+        sudo mv go  /usr/bin/
```

Comment:
1. Wouldn't `/usr/local/bin` be the more normal choice?  Or even `$HOME/local/bin`?
2. When _I_ `tar -xvf` that file, I do not get anything named `go`:

```console
(base) mspreitz@mjs12 test4 % ls -l
total 296960
-rw-r--r--  1 mspreitz  staff  151758876 Apr 17 21:55 go1.19.2.darwin-amd64.pkg

(base) mspreitz@mjs12 test4 % tar -xvf go1.19.2.darwin-amd64.pkg
x org.golang.go.pkg
x org.golang.go.pkg/Bom
x org.golang.go.pkg/Payload
x org.golang.go.pkg/Scripts
x org.golang.go.pkg/PackageInfo
x Resources
x Resources/bg-light.png
x Resources/bg-dark.png
x Distribution

(base) mspreitz@mjs12 test4 % ls -la
total 296968
drwxr-xr-x    6 mspreitz  staff        192 Apr 17 21:57 .
drwxr-xr-x+ 119 mspreitz  staff       3808 Apr 17 21:57 ..
-rwxr-xr-x    1 mspreitz  staff       1637 Apr 17 21:57 Distribution
drwx------    4 mspreitz  staff        128 Dec 31  1969 Resources
-rw-r--r--    1 mspreitz  staff  151758876 Apr 17 21:55 go1.19.2.darwin-amd64.pkg
drwx------    6 mspreitz  staff        192 Dec 31  1969 org.golang.go.pkg
```